.github/workflows/backend-ci.yml

```
@@ -0,0 +1,61 @@
+# This workflow will do a clean installation of python dependencies, cache/restore them, build the source code and run tests across different versions of python.
+# It uses Poetry to manage dependencies and pytest to run tests.
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Backend CI
+
+on:
+  workflow_dispatch:
+  pull_request:
```

You should add push trigger to run the ci on all branches

---

I did have a version with `push`. See https://github.com/dataforgoodfr/13_brigade_coupes_rases/pull/49/commits/cfbf60ffd99c547ab6c4e458280468fb57793ca3
The problem is that this will launch twice for any PR targeting main (once for the `push` event and once for the `pull_request` event).

---

If I leave it just on pull_request, we just launch the CI if the pull_request is created.

---

Yes I know but is it really a problem to launche twice ? I think it's better to fail fast on the CI instead of failing at the pull request time 😊

---

I agree, my only concern is that we have a limit on the CI (amount of free Actions minutes/month)... So maybe not efficient to launch it and use twice every time

---

Found a solution! --> https://github.com/dataforgoodfr/13_brigade_coupes_rases/pull/49/commits/258c1ad0ec3c7f1106de4f14a00fe7edea264337
It will launch on every push and if it comes from a PR --> launch once and cancel the other. 
Thanks for the feedback @sbailleul